tasks/deprecated-image-check.yaml

```diff
@@ -0,0 +1,72 @@
+apiVersion: tekton.dev/v1beta1
+kind: Task
+metadata:
+  name: deprecated-image-check
+spec:
+  stepTemplate:
+    volumeMounts:
+      - name: data
+        mountPath: /data
+
+  params:
+    - name: registry
+      description: "Image registry for the tested image"
+      default: registry.access.redhat.com
+    - name: source-image
+      description: "Image name to test"
+    - name: policy-dir
+      description: "Path to the directory containing conftest policies"
+      default: "/project/image/policy/deprecated-image.rego"
+
+  results:
+    - name: pyxis-http-code
+      description: HTTP code returned by Pyxis API endpoint
+
+  steps:
+    # Download Pyxis metadata about the image
+    #
+    # project
+    # └── policy
+    # data
+    # └── image_data.json
+    - name: query-pyxis
+      image: registry.access.redhat.com/ubi8/ubi
+      script: |
+        #!/usr/bin/env bash
+        echo "Querying Pyxis for $(inputs.params.source-image)..."
+        http_code=$(curl -s -o /data/image_data.json -w '%{http_code}' "https://catalog.redhat.com/api/containers/v1/repositories/registry/$(inputs.params.registry)/repository/$(inputs.params.source-image)")
+
+        echo "Response code: $http_code"
+        echo -n $http_code > $(results.pyxis-http-code.path)
+
+    # Run the tests and save output
+    #
+    # project
+    # └── policy
+    # data
+    # ├── image_data.json
+    # └── test_result.json
+    - name: run-conftest
+      image: quay.io/redhat-appstudio/hacbs-test
+      script: |
+        #!/usr/bin/env sh
```

Comment:
We should move this If/else logic out of this step and push it up into the first step where we do the query of pyxis. Right now it makes the most sense to the keep the pyxis logic together. That way this step emulates like the label step and just performs the conftest command. Thoughts?